Android settings: Develop adaptive HUD scaling

paramat opened the pull request:
Android settings: Develop adaptive HUD scaling

Use font size 14 for phones.
Use x_inches < 3.7 instead of < 3.5 for small phones.
Add a new category x_inches < 6 for larger phones.
Use HUD scaling 0.85 for larger phones.
Use desktop defaults for tablets.
//////////////////

For #5054 #6725 #7504 
To be updated after merge of #7834

stujones11 commented:
Works okay on my device (SG Tab-A)

![settings](https://user-images.githubusercontent.com/3710749/46917349-0c2fdf80-cfbe-11e8-984d-6116ff26ed4f.png)
![controls](https://user-images.githubusercontent.com/3710749/46917354-223da000-cfbe-11e8-8333-c8c77121b5d9.png)

The in-game controls do appear to be slightly smaller than before which causes some slight distortion of buttons, however, I think I can live with that.

![down](https://user-images.githubusercontent.com/3710749/46917362-3bdee780-cfbe-11e8-8b08-4ef109d3b8d8.png)

I still thing that the scaling algo could be improved to better take into account higher dpi displays but that is difficult to get right without access to large variety of devices to test on. Hopefully this is sufficient to fix the immediate user-facing issue well enough for a release.

Debug APK here: https://nofile.io/f/7ncYVYKVX7z/Minetest-debug.apk

MoNTE48 commented:
> The in-game controls do appear to be slightly smaller than before which causes some slight distortion of buttons
Yes, the buttons in MultiCraft are larger in code. For this, PR is probably worth reducing the size of the control buttons slightly.
Minetest: https://github.com/minetest/minetest/blob/master/src/gui/touchscreengui.cpp#L548
MultiCraft: https://github.com/MultiCraftProject/MultiCraft/blob/master/src/touchscreengui.cpp#L311
_The bigger, the better :)_
I tested the size of the control buttons on a huge list of devices. @paramat, you can take sizes from me if you want.

paramat commented:
Hmm perhaps HUD scaling needs to be slightly larger then for MT, i'd prefer to keep control size unchanged.

stujones11 commented:
@paramat My device is far from typical and I have never had a problem with gui scaling. I think that the important thing is that the UI is always at least useable enough to alter the settings to your own personal preference. We really need some feedback from mobile phone users to know if this works.

Increasing the base resolution of the control images would result in less distortion at the expense of memory, however, I think that cost is small enough to be justified in this case.

> i'd prefer to keep control size unchanged.

That is unlikely to be possible for everyone, only those that would get the current default values.

paramat commented:
Updated to 0.6/0.7.

MoNTE48 commented:
![46917349-0c2fdf80-cfbe-11e8-984d-6116ff26ed4f](https://user-images.githubusercontent.com/6764600/46922025-090b1280-d003-11e8-9c1f-500f4d8ffd43.png)
Offtop, however it is terrible.

stujones11 commented:
> This would still address the issues and seems suitable due to MT's smaller control buttons.

That might fix that problem for some but I would be careful adjusting these before we have more feedback from those that reported problems. Note that increasing the base resolution does not necessarily need to alter the actual size of the control, it just gives a better quality result, much like the inventory cube.

paramat commented:
Thanks for the apk, as it is currently seems a good place to start.
@IhrFussel @numberZero @codexp @rubenwardy @srifqi please could you test?

MoNTE48 commented:
> please could you test?

If you want a good test, you can always contact me. I have a large number of Android devices.

IhrFussel commented:
On my phone (which got a 5 inch screen but very low resolution like almost every cheap device on the market) it's still completely wrong:

![apktest](https://user-images.githubusercontent.com/22120097/46962308-1677d980-d0a3-11e8-8649-4bee54ae5f3e.png)

codexp commented:
Here is how it looks like on Huawei P10 lite:
![whatsapp image 2018-10-15 at 20 46 38 1](https://user-images.githubusercontent.com/6502486/46971706-11c01f00-d0bd-11e8-8798-6683ed0d8efc.jpeg)
![whatsapp image 2018-10-15 at 20 46 38 2](https://user-images.githubusercontent.com/6502486/46971707-11c01f00-d0bd-11e8-99f7-d5f065da8b20.jpeg)
![whatsapp image 2018-10-15 at 20 46 38 3](https://user-images.githubusercontent.com/6502486/46971708-11c01f00-d0bd-11e8-9db5-089fe0a6b931.jpeg)
![whatsapp image 2018-10-15 at 20 46 38 4](https://user-images.githubusercontent.com/6502486/46971710-1258b580-d0bd-11e8-92da-e60c059fc2b8.jpeg)
![whatsapp image 2018-10-15 at 20 46 38 5](https://user-images.githubusercontent.com/6502486/46971711-1258b580-d0bd-11e8-97be-2a36a6d4bfd8.jpeg)
![whatsapp image 2018-10-15 at 20 46 38](https://user-images.githubusercontent.com/6502486/46971712-1258b580-d0bd-11e8-9f9b-dca28f317512.jpeg)

Unfortunately after reinstalling 4.17.x version, game does not start anymore!

For me it is important (likely for many users), that I can play on 4.1x servers. I was not able to connect to any server, since protocols are not compatible. The new version should be released under a new name!

codexp commented:
btw. there are same problems with GUI in 4.17.x version (looks same).

codexp commented:
For me it is important (likely for many users), that I can play on 0.4.1x servers. I was not able to connect to any server, since protocols are not compatible. The new version should be released under a new name!

codexp commented:
Gear button should be in line with other buttons:
![whatsapp image 2018-10-15 at 20 46 38](https://user-images.githubusercontent.com/6502486/46973803-93ff1200-d0c2-11e8-9e2e-34b5acbcd361.jpeg)

paramat commented:
@IhrFussel @codexp thanks, i know it's far from perfect, but is it usable and is it an improvement?
Concerning menus, the important thing is:
> I think that the important thing is that the UI is always at least useable enough to alter the settings to your own personal preference.

codexp,
> Unfortunately after reinstalling 4.17.x version, game does not start anymore:

That error is caused by using MTGame 5.0.0dev with MTEngine 0.4.x, you need to use a matching version of MTGame.
> For me it is important (likely for many users), that I can play on 0.4.1x servers. I was not able to connect to any server, since protocols are not compatible.

That's simply the result of the network compatibility breakage of MT 5.0.0, nothing we can do about that.

MoNTE48 thank you, any testing you can do will be much appreciated.

paramat commented:
Tagging more Android users to request testing of the apk linked in https://github.com/minetest/minetest/pull/7784#issuecomment-429662776

@waterel @lhofhansl @aerozoic @MrChiantos @serofir @alnixon @juozaspo @Elkien3 @Wayward1 @WSDguy2014 @MuhdNurHidayat @jondo @AntumDeluge @Azzeccagarbugli @nanepiwo @ensonic @Warr1024 @JakubVanek @Kalabasa

Please state your screensize if you test, and please test advanced settings to see if you are able to adjust the scaling settings.

paramat commented:
LOL it seems a disaster whatever we do :D

benrob0329 commented:
I think if the advanced settings forms weren't so big, the font sizes could be dropped slightly and it would all fit.

juozaspo commented:
Looks like working. But you have to uninstall old version of minetest first.
```
$ adb install Minetest-debug.apk 
Failed to install Minetest-debug.apk: Failure [INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package net.minetest.minetest signatures do not match the previously installed version; ignoring!]
$ #uninstall old minetest app using phone interface
$ adb install Minetest-debug.apk #works
```
Will post screenshots later.

paramat commented:
benrob0329 good idea.

paramat commented:
benrob0329 good point. Advanced settings has menus larger than all other menus, which are nicely matched. I might try to reduce the sizes.

For this PR i'm sceptical of increasing GUI scaling to 1.2, maybe i'll leave that unchanged at 1.1.

juozaspo commented:
Screenshots of initial test where old Minetest folder were renamed
![screenshot_20181016-072143](https://user-images.githubusercontent.com/6626242/46993695-4cf03b80-d119-11e8-82d9-770680225ba1.png)
![screenshot_20181016-072153](https://user-images.githubusercontent.com/6626242/46993706-5b3e5780-d119-11e8-98db-2882d3b39ead.png)
![screenshot_20181016-072201](https://user-images.githubusercontent.com/6626242/46993716-64c7bf80-d119-11e8-81c2-d34cb89678cc.png)
![screenshot_20181016-072210](https://user-images.githubusercontent.com/6626242/46993722-70b38180-d119-11e8-9da6-286e3f5d0f1a.png)
Main ui, some fonts might be too large
![screenshot_20181016-072240](https://user-images.githubusercontent.com/6626242/46993751-9d679900-d119-11e8-82e0-23473201e0ce.png)
Spawning in -256 * -236?
![screenshot_20181016-072548](https://user-images.githubusercontent.com/6626242/46993786-e4558e80-d119-11e8-9cae-441afb732bc5.png)
![screenshot_20181016-072621](https://user-images.githubusercontent.com/6626242/46993792-ee778d00-d119-11e8-9a3f-6a735af42cf0.png)
![screenshot_20181016-072633](https://user-images.githubusercontent.com/6626242/46993896-7fe6ff00-d11a-11e8-9fc8-2966989308b7.png)
Ingame ui looks ok, through debug font might be too large.
![screenshot_20181016-073105](https://user-images.githubusercontent.com/6626242/46993964-d81e0100-d11a-11e8-9e52-152969b4b6df.png)
The movement controls obstruct mining some times
![screenshot_20181016-073522](https://user-images.githubusercontent.com/6626242/46993997-01d72800-d11b-11e8-950a-44cfdfb0fe61.png)
Pause menu font large too.

Tests done while in default settings, old Minetest folder in storage renamed.

paramat commented:
I'm now working on making advanced settings menus smaller to help with this, but also because they are a bit of a mess and values are clipped when using default screensize on desktop (1024x600).

juozaspo commented:
My screen info (dpi, etc., copied from Screen Info app)
```
Model: HUAWEI NEM-L21
Android version: 7.0 - N
API level: 24
Drawable density: xxhdpi
Layout size: normal
Screen (dpi): 480
Width (px): 1080
Height (px): 1920
Usable width (px): 1080
Usable height (px): 1812
Width (dp): 360
Height (dp): 640
Smallest dp: 360
Orientation: Portrait
Orientation degree: 0°
Diagonal Size(inches): 5.0
X dpi: 422,0
Y dpi: 424,1
Touchscreen: Finger
Refresh rate (fps): 60,0
```

paramat commented:
Thanks. I think this PR will need some rethink. MultiCraft may have cusotmised menus so their GUI scaling of 1.2 may not be suitable for MT.

juozaspo commented:
Another quirk for this build (apk above): Reset singleplayer world doesn't do anything. Clicking yes or no in dialog doesn't change anything, still loading the same world :\
adding debug_log_level = verbose to minetest.conf shows interesting results in debug.txt:
```
2018-10-16 16:30:37: INFO[Main]: Removing "/storage/emulated/0/Minetest/worlds/singleplayerworld"
2018-10-16 16:30:37: VERBOSE[Main]: Executing '/bin/rm' '-rf' '/storage/emulated/0/Minetest/worlds/singleplayerworld'
2018-10-16 16:30:37: INFO[Main]: Initializing world at /storage/emulated/0/Minetest/worlds/singleplayerworld
2018-10-16 16:30:37: INFO[Main]: Searching worlds...
2018-10-16 16:30:37: INFO[Main]:   In /storage/emulated/0/Minetest/worlds: 
2018-10-16 16:30:37: INFO[Main]: singleplayerworld 
2018-10-16 16:30:37: INFO[Main]: 1 found.
```
Minetest is trying to run `/bin/rm` which probably does not exist. Checking for /bin/rm in adb shell finds nothing:
```
HNNEM-H:/ $ ls /bin/rm
ls: /bin/rm: No such file or directory
```

rubenwardy commented:
> But you have to uninstall old version of minetest first.

This is because the signing key changed because a different computer was used to make the ADB. This won't affect the official builds

MoNTE48 commented:
@paramat I forgot to mention that MC uses a different font, which is smaller by default. Perhaps it is worth set a smaller font size for small screens?

MoNTE48 commented:
![photo_2018-10-16_16-54-03](https://user-images.githubusercontent.com/6764600/47025640-27d4ea80-d164-11e8-8486-577964dafc85.jpg)
MultiCraft Credits tab. Oneplus 3 - 5.5" FullHD
Perhaps such an implementation will be more beautiful, as well as look better on both PCs and mobile devices with a small screen?

stujones11 commented:
> Looks like working. But you have to uninstall old version of minetest first.

Yes, this is a very important point, apologies for not making this clear. To ensure you get the new default settings you will need to completely remove any existing installation. This incudes removing, or better, renaming your Minetest folder.

@codexp you will need to do the same to recover your old 0.4.17 installation.

> LOL it seems a disaster whatever we do :D Android is such a pain.

Certainly it is challenging but other games seem to manage alright. From what I can see from most of these screenshots is the font is too large, maybe we also need to reduce that for smaller screen as well?

Another possibility would be to provide separate, more simplified menus for mobile.

> This is because the signing key changed because a different computer was used to make the ADB.

This is an **unsigned** debug build, so it has nothing to do with that.

MoNTE48 commented:
> Another possibility would be to provide separate, more simplified menus for mobile.

For settings - yes. And for singleplayer and multiplayer, it would be better to choose a standard PC menu. It is much more convenient and not so cluttered with buttons.

juozaspo commented:
While minetest app is running, removing `singleplayerworld` from minetest world directory causes this to show on next attempt to run singleplayer.
![screenshot_20181016-195158](https://user-images.githubusercontent.com/6626242/47033322-282daf80-d17d-11e8-9898-62847e2a56e4.png)

Next time world loads ok.

stujones11 commented:
@juozaspo Thank you for testing but please can you post a new issue for this.

> The Reset Singleplayer World button doesn't function properly

However, deleting the world while the app is running is never going to end well :D

srifqi commented:
Minetest tested on Samsung Galaxy J7 (5.5" display)

Default settings:
![default-mainmenu](https://user-images.githubusercontent.com/4017302/47036869-a56e1a00-d1a7-11e8-9e36-250e20702b33.png)
![default-settings](https://user-images.githubusercontent.com/4017302/47036870-a606b080-d1a7-11e8-8cf1-d8a5fbf08741.png)
![default-advancedsettings](https://user-images.githubusercontent.com/4017302/47036871-a606b080-d1a7-11e8-8623-f1b837c50b85.png)
![default-advancedsettingentry](https://user-images.githubusercontent.com/4017302/47036873-a69f4700-d1a7-11e8-942c-5f6d0c9971f9.png)

`gui_scaling = 0.9`
![90percentguiscale-advancedsettingentry](https://user-images.githubusercontent.com/4017302/47036956-dd755d00-d1a7-11e8-970d-79241721705a.png)
![90percentguiscale-settings](https://user-images.githubusercontent.com/4017302/47036958-de0df380-d1a7-11e8-95c3-465b9f91902f.png)
![90percentguiscale-ingame](https://user-images.githubusercontent.com/4017302/47036960-dea68a00-d1a7-11e8-8e79-ba82ad169b7a.png)

stujones11 commented:
@paramat Based on what I have seen so far, the gui-scaling of 1.2 needs to be reduced to 0.9 for anything under 6 inches. This would mean an additional case for larger screen devices (like my tablet) otherwise the menus will be ridiculously small. We probably also need to adjust the default font size accordingly.

juozaspo commented:
@stujones11 @paron my Huawei phone I have gui scaling as 0.85 and fonts reduced to 11, setting gui scale to 0.9 the dialog is a bit too large. 
![screenshot_20181016-215931](https://user-images.githubusercontent.com/6626242/47041120-a5aeeb00-d190-11e8-97f6-bbc114c682a5.png)
![screenshot_20181016-220003](https://user-images.githubusercontent.com/6626242/47041128-ae9fbc80-d190-11e8-8120-23952da4d770.png)
![screenshot_20181016-220242](https://user-images.githubusercontent.com/6626242/47041144-b5c6ca80-d190-11e8-88f9-a3652e5ad27c.png)
The screen dpi x is not the same as y, over both over 420, diagonal 5.0 inches. Look at top of https://github.com/minetest/minetest/pull/7784#issuecomment-430131919

stujones11 commented:
> setting gui scale to 0.9 the dialog is a bit too large.

IMO this particular dialog is over-sized on any device (including PC). There is very much empty space.

paramat commented:
> Based on what I have seen so far, the gui-scaling of 1.2 needs to be reduced to 0.9 for anything <= 5.5 inches.

For advanced settings edit formspec yes, but that would then make other menus much smaller, causing other problems.
I'll minimise the size of the advanced setting edit formspec in my upcoming PR.
I agree that all screenshots i've seen seem to have font too large.

Eventually, perhaps Android needs a dedicated basic settings menu that makes the critical settings GUI scaling, HUD scaling and font size quickly and easily accessible.

paramat commented:
I have an idea.
It seems that GUI scaling 1.2 is helpful, the main problem here is the oversized advanced settings editing formspec.
I might alter the code such that the formspec for editing simple settings (GUI/HUD scaling, font size) is as small as possible, then this can be a short term solution to at least make it possible to edit these values on any Android device.
There are formspecs for editing noise params, mapgen flags etc. which need to be large so i'll leave those unchanged.
This means this PR could perhaps go ahead in it's current form.

Additionally we could perhaps set a slightly smaller font size for Android, but only slightly to not then cause too-small fonts on some devices. The large fonts are ugly but at least they are readable and usable.

paramat commented:
See PR #7791 perhaps it helps.

paramat commented:
But, does 1.2 help with opening up sections of the advanced settings menu tree?

paramat commented:
Closing, as increasing GUI scaling makes large formspecs worse and there's no evidence the changes to HUD scaling help MT. MultiCraft settings are not necessarily best for MT due to their custom menus and differently sized controls.
I might resurrect this in future if any helpful change becomes clear.

stujones11 commented:
> Closing, as increasing GUI scaling makes large formspecs

I think this change is well worth keeping, we just need to find the default values best suited to MT which (IMO) should also include the font size. I would also like to have a check included for larger screens, like a tablet, that behaves more like a small pc monitor.

> there's no evidence the changes to HUD scaling help MT.

https://github.com/minetest/minetest/issues/7504#issuecomment-401634375

paramat commented:
Hmm ok #7504 is a worthwhile change, and i can add a check for large tablets to use desktop settings. At which screen size of tablet would you suggest?

stujones11 commented:
@paramat Thank you, I was thinking that anything 7 inches or above should be okay with the defaults but this can always be tweaked if necessary. IMO, the important thing is that we keep the possibility of adjusting defaults based on physical screen size.

While the improvements to the advanced setting dialogs will, hopefully, fix the blocking issue of users being unable to alter the settings, I still think it would be nice give new users the best possible first time experience.

paramat commented:
Done.
Thanks for the encouragement.
For larger phones 0.85 sits nicely between 0.7 and 1.0, and may help with #7504 type issues.

12Me21 commented:
Wait, so x_inches is the width of the screen in inches, right?

paramat commented:
Ok, can anyone clarify? I assumed 'inches' implied this is the width of the screen in inches. that `porting::getDisplaySize().X` gets width in pixels and `160 * porting::getDisplayDensity()` is DPI.

stujones11 commented:
@paramat, yeah sorry, I did make that mistake so it possibly could be reduced slightly. However, my own tablet actually measures just over 7 inches wide so is probably okay either way.

MoNTE48 commented:
@stujones11 What is your screen resolution?

stujones11 commented:
> What is your screen resolution?

1024 x 768 pixels, not sure about dpi but is obviously quite low. I can find out if necessary.

MoNTE48 commented:
Correct me. FullHD tablet, xxhdpi. Big screen, small resolution.
xxhdpi = 3.0
1920/(160*3.0)=4

stujones11 commented:
@MoNTE48 I am sure, it's about 3 years old so maybe not the best.

Here are the manufacturer's specs: https://www.samsung.com/us/mobile/galaxy-tab/SM-T550NZAAXAR-specs

stujones11 commented:
@MoNTE48 Yes, what I meant was the `x_inches` would probably be just over 7 inches it the formula is good. That is what it measures in width.

MoNTE48 commented:
@stujones11 1280/(160*1.5)=5.33...
Is hud_scaling 0.7 not too little for you? Can I see a screenshot?

stujones11 commented:
Is hud_scaling 0.7 not too little for you?

Yes it is, 0.85 is acceptable but it is best left at the default 1.0. That is why I requested a higher value with no changes at all. I did, however, initially think `x_inches` was the screen-size, though it should have been obvious by the 'x'

Screenshot with `hud_scaling = 0.85` https://github.com/minetest/minetest/pull/7784#issuecomment-429626965

MoNTE48 commented:
@paramat 
	if (x_inches <= 3.7) {
		// small 4" phones
		settings->setDefault("hud_scaling", "0.5");
		settings->setDefault("gui_scaling", "1.1");
		settings->setDefault("mouse_sensitivity", "0.25");
	} else if (x_inches > 3.7 && x_inches < 5) {
		// all phones
		settings->setDefault("hud_scaling", "0.6");
		settings->setDefault("gui_scaling", "1.1");
		settings->setDefault("mouse_sensitivity", "0.15");
	} else if (x_inches >= 5) {
		// tablets
		settings->setDefault("hud_scaling", "0.85");
		settings->setDefault("gui_scaling", "1.1");
	}

stujones11 commented:
@MoNTE48 That looks about right to me, although I still think that we need to reduce font size for smaller displays.

MoNTE48 commented:
@stujones11 
settings->setDefault("font_size", font_size_str"-2"); ?
Not sure how you need it, please tell me.

stujones11 commented:
@MoNTE48 I don't need it but judging by the some of the screen shots in this issue (and elsewhere) it would probably be beneficial for smaller screen phones. Unfortunately my phone is on Android 2.2 so no go there. I really should get myself a new one :D

stujones11 commented:
@MoNTE48 From what I can tell the default font size is 18 or `"18"` in string form. iPhone is a bit pricey for me and I personally dislike the company. My main reason for not upgrading is that I refuse to be tied to a contract. Anything I do get will have to be SIM free.

MoNTE48 commented:
@stujones11, I think that paramat will want to do it more beautifully, and not hard-code font size. 
We need to get a standard font size **font_size_str** (because it may be changed in the future) and reduce or increase it.

_Yes, only SIM free! Doesn't your country sell such phones? (You can love Apple, or you can not understand how terrible Android is. Holy War!!!)_

stujones11 commented:
> I think that paramat will want to do it more beautifully

I suppose it could be included in the constants as it is done elsewhere. @paramat can decide how best to implement it.

_I guess iPhones are probably very good, I can only really speak for their horrible desktop OS I have to suffer at work, it's even worse than Windows :))_

paramat commented:
>  In this case 6 inches may better as the upper limit

I'll change the 7 to a 6.
As for everything else, i have little idea what will work. However it seems a slight font size reduction might help, looking at the screenshots.

rubenwardy commented:
This improves the in-game HUD and makes the advanced settings menu more usable.

There's still some issues with font size:

![screenshot_20181024-102752](https://user-images.githubusercontent.com/2122943/47421225-0d100080-d778-11e8-9bb0-c69a94793e53.jpg)


This is with a s7 edge: 1440x2560 @ 5.5" = 534 DPI

rubenwardy commented:
Please make sure you include your resolution and DPI with any findings

paramat commented:
I possibly could set font size smaller for smaller screens in this PR.

paramat commented:
7.0 changed to 6.0.

paramat commented:
Updated to set font_size 14 for phones.
Removed redundant setting of settings for >= 6", replaced with comment.

According to
https://github.com/minetest/minetest/blob/b298b0339c79db7f5b3873e73ff9ea0130f05a8a/src/constants.h#L115
and
https://github.com/minetest/minetest/blob/18a8fbf465b47c4780d63f34c960af7a52d0cf82/src/defaultsettings.cpp#L264
Font size may be 16 or 10 depending on `USE_FREETYPE`, does Android always use `USE_FREETYPE`?

And does `mono_font_size` need setting too?

MoNTE48 commented:
> Android **always** use USE_FREETYPE
> And does mono_font_size need setting too?
This font can be used for some languages.

paramat commented:
Updated to use 3.7 and also set 'mono font size'.
Thanks MoNTE48 i appreciate your help on this.

stujones11 commented:
@paramat this looks like a good staring point, although I have a feeling that the font sizes should decrease with the display size, maybe 15, 14, 13? I am only really guessing here so maybe await more user feedback.

paramat commented:
Thanks. I'm beginning to think GUI scaling should be left at 1 for all screen sizes?

stujones11 commented:
> Thanks. I'm beginning to think GUI scaling should be left at 1 for all screen sizes?

Maybe for `x_inches < 6.0f` but 1.1 is just fine on my 9.7 inch tablet, though I guess that would not be affected anyway,

paramat commented:
Updated.

stujones11 commented:
> Updated.

LGTM :+1:

stujones11 commented:
@paramat Sorry to be a pain, however, if #7834 is merged then the further reduction in font size might not be required. My device would be fine with font-size 14 and it does make things like the advanced tree-view much easier to use. The `gui_scaling` lines will also likely become redundant.

paramat commented:
Hey no problem, it's good to see your PR. Let's get that merged first then you can let me know the new optimum form of this PR.

stujones11 commented:
From my own experience and the comments in #7834 I now think is better to just leave the font size at 14 as it was before. I think it is probably best to err on the large side otherwise the advanced menu tree-view can become very difficult to navigate.

I don't think the `gui_scaling` lines should be needed now the scaling is handled automatically by the engine. Otherwise, I see no reason why this should not be merged.

paramat commented:
Thanks will update.

paramat commented:
Updated, but applies font size 14 for tablets (> 6), hope this is ok (can't judge).

stujones11 commented:
> Updated, but applies font size 14 for tablets (> 6), hope this is ok (can't judge).

That was actually better left at 16 for my own device. Like I say, tablets tend to behave a lot more like desktop monitor. Size 14 on my old `ldpi` tab-a is quite difficult to read and suffers the same problem I mention with the advanced settings tree-view.

paramat commented:
Thanks, updated, look ok?

stujones11 commented:
This looks good to me know, thank you for your patience :+1:
